internal/gatewayapi/globalresources.go

```diff
@@ -28,17 +43,58 @@ func (t *Translator) ProcessGlobalResources(resources *resource.Resources, xdsIR
 
 	for _, xdsIR := range xdsIRs {
 		if containsGlobalRateLimit(xdsIR.HTTP) || containsWasm(xdsIR.HTTP) {
-			xdsIR.GlobalResources = &ir.GlobalResources{}
+			if xdsIR.GlobalResources == nil {
+				xdsIR.GlobalResources = &ir.GlobalResources{}
+			}
 			xdsIR.GlobalResources.EnvoyClientCertificate = &ir.TLSCertificate{
 				Name:        irGlobalConfigName(envoyTLSSecret),
 				Certificate: envoyTLSSecret.Data[corev1.TLSCertKey],
 				PrivateKey:  envoyTLSSecret.Data[corev1.TLSPrivateKeyKey],
 			}
 		}
 	}
+
 	return nil
 }
 
+// processServiceClusterForGateway returns the matching IR key for a gateway and builds a RouteDestination to represent the ProxyServiceCluster
+func (t *Translator) processServiceClusterForGateway(gateway *GatewayContext, resources *resource.Resources) (string, *ir.RouteDestination) {
+	var (
+		labels = make(map[string]string)
+		irKey  string
+	)
+
+	if t.MergeGateways {
+		irKey = string(t.GatewayClassName)
+		labels[OwningGatewayClassLabel] = string(t.GatewayClassName)
+	} else {
+		irKey = irStringKey(gateway.Namespace, gateway.Name)
+		labels[OwningGatewayNamespaceLabel] = gateway.Namespace
+		labels[OwningGatewayNameLabel] = gateway.Name
+	}
+
+	svcCluster := resources.GetServiceByLabels(labels, t.ControllerNamespace)
```

Comment:
will this work for gateway namespace mode ? @zirain

Reply:
yeah, we should get from gateway namespace in gateway namspace mode.

Reply:
What should be used when both merged gateways and gateway namespace mode are enabled?

Reply:
Should be resolved now